core/src/Threads/Kokkos_Threads_Parallel_Range.hpp

```diff
@@ -249,43 +234,28 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
         const pointer_type data =
             (pointer_type)ThreadsExec::root_reduce_scratch();
 
-        const unsigned n = Analysis::value_count(
-            ReducerConditional::select(m_functor, m_reducer));
+        const unsigned n = reducer.value_count();
         for (unsigned i = 0; i < n; ++i) {
           m_result_ptr[i] = data[i];
         }
       }
     }
   }
 
-  template <class HostViewType>
-  ParallelReduce(const FunctorType &arg_functor, const Policy &arg_policy,
-                 const HostViewType &arg_result_view,
-                 std::enable_if_t<Kokkos::is_view<HostViewType>::value &&
-                                      !Kokkos::is_reducer<ReducerType>::value,
-                                  void *> = nullptr)
-      : m_functor(arg_functor),
+  template <class ViewType>
```

Comment:
The renaming of the template parameter was not a good change.  Not blocking but I don't like it.

Reply:
The point of the improved `static_assert` was that the View doesn't need to live in `HostSpace` but just in a space that is host-accessible.

Reply:
And the "host" prefix convey the idea.

Reply:
Actually I am kinda ok with the change. The HostViewType could be read as it needs to be a litteral HostSpace thing, and wouldn't allow for SharedSpace for example?